src/service/engine/index.ts

```diff
@@ -90,41 +96,60 @@ export class EngineService {
     return engines;
   }
 
+  private async validateCreateOptions(
+    accountVersion: number,
+    options: CreateEngineOptions
+  ) {
+    const disallowedV2Options = ["region", "engine_type", "warmup"];
+    if (accountVersion >= 2) {
+      // find a list of disallowed options that are set and report them in an exception
+      const disallowedOptions = disallowedV2Options.filter(
+        option => (options as Record<string, string>)[option] !== undefined
+      );
+      if (disallowedOptions.length > 0) {
+        throw new DeprecationError({
+          message: `The following engine options are not supported for this account: ${disallowedOptions.join(
+            ", "
+          )}`
+        });
+      }
+    } else if (options.engine_type == undefined) {
+      options.engine_type = EngineType.GENERAL_PURPOSE;
+    }
+    if (options.fail_if_exists == undefined) {
+      options.fail_if_exists = true;
+    }
```

Comment:
I think it's best to refactor setting defaults in their own function. This way it's better logically, where one validates another one sets the defaults. Otherwise, personally, it's a bit unexpected that validation step also modifies the values.